interview-questions.md

```diff
@@ -8,36 +8,61 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: You can fix this by migration. The name of the foreign key would be student_id, and it would be on the student model. This is because the cohort model already has an id that the many students would have to match up with.
 
-Researched answer:
+Researched answer: Migration can fix this issue. You will need to change the student model to include the foreign key, which in this case would be student_id. The cohort model already has a built in id that once you add the student id, because you already have it set to the cohort has_many students, it will automatically associate those things when you add the student id. Once you add that, you can migrate it and it will save the student model with the new student_id tag.
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: Create, Update, Show, and Destroy. This is because for create you need to have the params for what you are creating, aka a title or a date, and for update you need to see those same params to change them. For show you need an ID number so that you can find that one instance, and for Destroy you also need an ID number so you know which instance to delete.
 
-Researched answer:
+Researched answer: The restful routes that must always be passed a param is Create, Update, Show, Destroy, and Edit. These all need params so that, in the case of Edit and Show, we know which instance to either show on our page or display so we can edit it. In the case of destroy, we need the ID of that instance to know which one to delete. And in the case of Create and Update, we need to know what to replace the already written model in our database with. 
```

Comment:
Review this diagram to get more clarity. You should see the four that need the id param